utilities/src/main/java/io/syndesis/qe/camelk/CamelKTemplate.java

```diff
@@ -0,0 +1,163 @@
+package io.syndesis.qe.camelk;
+
+import static org.assertj.core.api.Fail.fail;
+
+import io.syndesis.qe.TestConfiguration;
+import io.syndesis.qe.endpoints.TestSupport;
+import io.syndesis.qe.utils.OpenShiftUtils;
+import io.syndesis.qe.wait.OpenShiftWaitUtils;
+
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.io.FileUtils;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import lombok.Data;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class CamelKTemplate {
+
+    @Data
+    private static final class CamelKConfig {
+        public final String enabled;
+    }
+
+    @Data
+    public static final class CamelKIntegration {
+        public final String id;
+        public final String phase;
+        public final String context;
+    }
+
+    public static void editSpec(Map<String, Object> spec) {
+        Map<String, Object> addons = (Map<String, Object>) spec.computeIfAbsent("addons", s -> new HashMap<String, Object>());
+        addons.put("camelk", new CamelKConfig("true"));
+    }
+
+    public static void deployCamelKOperator() {
+        ProcessBuilder pb = new ProcessBuilder("docker", "pull", TestConfiguration.getCamelKOperator());
+
+        try {
+            pb.start().waitFor();
+        } catch (IOException | InterruptedException e) {
+            log.error("Unable to pull Camel K operator \"{}\"", TestConfiguration.getCamelKOperator(), e);
+        }
+
+        downloadArchive();
+        extractArchive();
+        installViaBinary();
+    }
+
+    private static final String CAMEL_K_ARCHIVE_PATH = String.format(
+        "https://github.com/apache/camel-k/releases/download/%s/camel-k-client-%s-%s-64bit.tar.gz",
+        "0.3.4",
+        "0.3.4",
+        System.getProperty("os.name").toLowerCase().contains("mac") ? "mac" : "linux"
+    );
+    private static final String LOCAL_ARCHIVE_PATH = "/tmp/camelk.tar.gz";
+    private static final String LOCAL_ARCHIVE_EXTRACT_DIRECTORY = "/tmp/camelk";
+
+    public static void deploy() {
+        downloadArchive();
+        extractArchive();
+        installViaBinary();
+        OpenShiftUtils.binary().execute("secrets", "link", "camel-k-operator", "syndesis-pull-secret", "--for=pull");
+        OpenShiftUtils.binary().execute("delete", "pod", "-l", "name=camel-k-operator");
```

Comment:
I know its easier to do it with binary client, but the preferred way should be using the java client